### PR TITLE
Move to a pyproject.toml based setup

### LINKED ISSUE #81: Pin numpy < 2.0
NumPy 2 will arrive Dec/Jan and NumPy have recommended pinning and testing against its release candidate(s) before using. Other Equinor packages are taking this action as well.

### LINKED ISSUE #86: Move to a pyproject.toml based package
See examples in subscript, rms-sys, etc.

### LINKED ISSUE #87: Add linting to CI workflows and lint code
This does not seem to be currently done. With black, flake8, isort, etc.

### LINKED ISSUE #99: Run CI nightly
```yml
  schedule:
    # Run nightly to check that tests are working with latest dependencies
    - cron: "0 0 * * *"
 ```

### PR DESCRIPTION
**Pull request by mferrera**
Resolves #81
Resolves #86 
Resolves #87 
Resolves #99
Does half of #95

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,36 +1,111 @@
...
+    "xtgeo",
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Science/Research",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Physics",
+    "Topic :: Software Development :: Libraries",
+    "Topic :: Utilities",
+    "Operating System :: POSIX :: Linux",
```
Comment (jcrivenaes) [inline pyproject.toml:36]: Works on Macs and Windows also, I would think

File: pyproject.toml [hunk 2]
```diff
@@ -1,36 +1,111 @@
...
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
```
Comment (jcrivenaes) [inline pyproject.toml:39]: 3.11 also?

**Review by jcrivenaes — APPROVED** [review]
Great work (a few minor comments only)

File: pyproject.toml (hunk 1 above)
Comment (mferrera) [reply pyproject.toml:36]: Good catch, added

**Pull request merged**